prov/sockets/src/sock_progress.c

```diff
@@ -1177,21 +1181,30 @@ ssize_t sock_rx_claim_recv(struct sock_rx_ctx *rx_ctx, void *context,
 	return ret;
 }
 
-static int sock_pe_progress_buffered_rx(struct sock_rx_ctx *rx_ctx)
+static int sock_pe_progress_buffered_rx(struct sock_rx_ctx *rx_ctx,
+					bool shallow)
 {
 	struct dlist_entry *entry;
 	struct sock_pe_entry pe_entry;
 	struct sock_rx_entry *rx_buffered, *rx_posted;
 	size_t i, rem = 0, offset, len, used_len, dst_offset, datatype_sz;
+	size_t max_cnt = SOCK_EP_MAX_PROGRESS_CNT;
```

Comment:
Avoid always setting max_cnt here, since we'll just change it if shallow is true.  Move to an else statement below.

Reply:
fixed.